assembly/src/assembler/mod.rs

```diff
@@ -300,27 +304,35 @@ impl Assembler {
         &self,
         proc_id: &ProcedureId,
         context: &mut AssemblyContext,
-    ) -> Result<&Procedure, AssemblyError> {
-        // if the procedure is already in the procedure cache, return it
-        if let Some(p) = self.proc_cache.get(proc_id) {
-            return Ok(p);
+    ) -> Result<impl Deref<Target = Procedure> + '_, AssemblyError> {
```

Comment:
I think the code here looks a bit more complicated than I would like (at least for a novice like myself). So, I would still prefer to get rid of `get_imported_proc()` and go with the `ensure_prcedure_in_cache()` approach.

Reply:
This `impl _` is useful when you don't want to specify the returned type but instead a behavior. The intent of the syntax is to actually simplify, so if it instead makes things more unreadable, it should be entirely removed since it will not fit its purpose.

Removed!